Add ark plugins (#1527)
kyma-project/kyma: tools/ark-plugins/README.md

# Ark plugins

## Overview

> Ark plugins provide an implementation of a few plugins required to properly restore kyma cluster and created resources inside it. It's mainly focused on resources related to the service catalog (instances, bindings, etc). Each plugin is defined in separate file inside [internal/plugins](internal/plugins) folder. Purpose of each plugin is defined in the file's comments.

Reviewer: ```suggestion
Ark plugins provide an implementation of a few plugins required to properly restore the Kyma cluster and resources created inside it. They focus mainly on resources related to the Service Catalog, such as instances or bindings. Each plugin is defined in a separate file inside the [`internal/plugins`](internal/plugins) folder. The purpose of each plugin is defined in the file's comments.
```

> Structure of the folders/files is based on [ark plugin example repository](https://github.com/heptio/ark-plugin-example).

Reviewer: ```suggestion
The structure of folders and files is based on the [Ark plugin example repository](https://github.com/heptio/ark-plugin-example).
```

## Installation

> The docker image build from [Dockerfile](Dockerfile) is run as an init container inside deployment of [ark chart](../../resources/ark). During development, you can push built image to your own image repository and run `ark plugin add [yourRepo/imageName:tag]`.

Reviewer: ```suggestion
Run a Docker image build from the [Dockerfile](Dockerfile) as an init container inside the deployment of the [Ark chart](../../resources/ark). During development, you can push a built image to your own image repository and run this command:
```

Reviewer: Please add the command as a code block:
```
ark plugin add {yourRepo/imageName:tag}
```

# Ark plugins

## Overview

Ark plugins provide an implementation of a few plugins required to properly restore the Kyma cluster and resources created inside it. They focus mainly on resources related to the Service Catalog, such as instances or bindings. Each plugin is defined in a separate file inside the [`internal/plugins`](internal/plugins) folder. The purpose of each plugin is defined in the file's comments.

The structure of folders and files is based on the [Ark plugin example repository](https://github.com/heptio/ark-plugin-example).

## Installation

> The docker image build from [Dockerfile](Dockerfile) is run as an init container inside deployment of [ark chart](../../resources/ark). During development, you can push built image to your own image repository and run:

Reviewer: ```suggestion
Run a Docker image build from the [Dockerfile](Dockerfile) as an init container inside the deployment of the [Ark chart](../../resources/ark). During development, you can push a built image to your own image repository and run:
```
